internal/frontend/middleware/global.go

```diff
@@ -95,15 +98,21 @@ func Setup(opts *Options) {
 	authBasic = opts.AuthBasic
 	authToken = opts.AuthToken
 	appLogger = opts.Logger
+	basePath = opts.BasePath
 }
 
 func prefixChecker(next http.Handler) http.Handler {
 	return http.HandlerFunc(
 		func(w http.ResponseWriter, r *http.Request) {
-			if strings.HasPrefix(r.URL.Path, "/api") {
-				next.ServeHTTP(w, r)
+			if basePath != "" && r.URL.Path == "/" {
+				http.Redirect(w, r, basePath, http.StatusSeeOther)
+				return
+			}
```

Comment:
If the root of the server is hit and we have a base path configured redirect to that base path for convenience.

Reply:
I've tested this locally with an nginx reverse proxy (configuration below) but encountered an issue when accessing via the proxy path `http://localhost/dagu/`. It seems the path prefix check might be the cause. The following code, however, works as expected:

```go
if strings.HasPrefix(r.URL.Path, "/api") {
    next.ServeHTTP(w, r)
} else {
    defaultHandler.ServeHTTP(w, r)
}
```

Would you mind taking a look when you have a chance?

Test setup
```nginx
server {
    listen       80;
    server_name  localhost;
    location /dagu/ {
        proxy_pass http://localhost:8080/;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}
```

~/dagu/admin.cfg
```yaml
basePath: /dagu/
port: 8080
```

Reply:
The issue is the trailing slash. I was only testing with `/dagu` and missed this case. 

I have made the BasePath more robust 

- Normalize the value in config.go so that all parts of the app can be sure it has the same shape
- changed `prefixChecker` to strip the prefix first, then do the strings.HasPrefix check in a HandlerFunc to not have to worry about the prefix. 

I tested with 

- no basepath set at all
- basepath set as an empty string
- basePath set as `/`
- basePath set as `/dagu`
- basePath set as `/dagu/`
- basePath set as `/dagu/foo`

I believe this should be good. 

I have pushed up the fix as a `!fixup` commit to make it easier to review. This can be "squashed" when merging or I can squash it before this is ready to merge.